Geometry/CaloGeometry/interface/CaloGeometry.h

```diff
@@ -31,10 +32,10 @@ class CaloGeometry
 			      const CaloSubdetectorGeometry* geom    ) ;
     
       /// Get the position of a given detector id
-      const GlobalPoint& getPosition( const DetId& id ) const ;
+      const GlobalPoint& getPosition( const DetId& id ) const;
 
       /// Get the cell geometry of a given detector id
-      const CaloCellGeometry* getGeometry( const DetId& id ) const ;
+      std::shared_ptr<const CaloCellGeometry> getGeometry( const DetId& id ) const;
```

Comment:
@bsunanda - there is no need for shared pointer here since nothing will share the ownership of it with CaloGeometry.

Reply:
CaloCellGeometry is recommended to be addressed using shared_ptr<const CaloCellGeometry> everywhere.